### PR TITLE
Proof of Concept: introducing Static Code Analysis, PHPStan and refactor for level 0

### PR DESCRIPTION
**Pull request by roberto-butti**
## Pull request type

Jira Link: no internal

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Other (please describe): Proof of concept, for introducing PHPStan and all refactor needed for reaching Level 0 (the minimum)

## How to test this PR

- composer install (for installing phpstan)
- vendor/bin/phpstan analyse src --level=0
-

## What is the new behavior?
Due to the changes needed for reaching level 0 of a static code analysis, i'm creating a PR, as a Proof of Concept , just to demonstrate that introducing tools like phpstan is feasible, but some refactor in the code is needed.
It is ok if you don't want to merge it, because probably you want to evaluate deeper and better the refactor needed.
The refactor was:
- moving functions that call getStories() from BaseClient to Client;
- moving "enrich" related functions from BaseClient to Client
- moving "relations" related functions from BaseClient to Client
- moving "resolvingLinks" related functions from BaseClient to Client
- declare attributes for classes
- cleaning responseHandler() for avoid duplicate code in the classes

I have some proposal (not yet included in this PR): creating Traits to group methods and attributes related to a specific functionality, for example:

- Traits  Relatable with: $_relationsList, getResolvedRelations() , insertRelations(), 
- Traits Enrichable with: enrichContent(), enrichStories()
- Traits Linkable with: getResolvedLinks(), insertLinks()

- 
- 
- 

## Other information

### CODE REVIEW

**Comment by pscheit** [comment]
Nice!

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,4 @@
+parameters:
+  level: 0
```
Comment (pscheit) [inline phpstan.neon:2]: I would suggest the following.

use: level: max here or lets say level 8 (9 is really hardcore)

and then create and include a baseline where we put all the issues into. Then we will shrink down the baseline issue by issue. That way new code is validated with full phpstan power.

Whole config would look like this:

```neon
includes:
    - phpstan-baseline.neon

parameters:
    level: 8
    
    paths:
        - src
       
```
You would generate the baseline with

`vendor/bin/phpstan analyse --generate-baseline=phpstan-baseline.neon`

and commit that

**Comment by pscheit** [comment]
btw i dont think moving those methods from the BaseClient to the Client works, since the Management Client relies on those methods as well?

**Comment by roberto-butti** [comment]
We will introduce PHP STAN as part of the refactoring process. Closing this PR for now

**Pull request closed without merging**